### ISSUE TITLE
RFC: Add vNext component manual accessibility testing process + wiki

### ISSUE THREAD

**Issue by smhigley** [microsoft/fluentui#20916]
## Problem

We want to catch any accessibility issues before we do official testing, both to save time and to prevent fixable issues going into RC that would require breaking changes.

## Proposed solution

1. We add this to the wiki as a template for a11y testing specific components: https://gist.github.com/smhigley/0f08c38758359e62f06d692a23860e63
2. When a component implementation is finished, the component author opens an a11y testing issue, using this as a template.
3. This should probably go in the component release checklist

I'm mostly interested in reviews of the testing template/checklist 😄. The idea is for component authors to do most of the review on their own, then either pair or hand it off for the more specialized parts, if needed (e.g. the screen reader testing and ARIA compliance).

I put the draft in a gist for now, since it's not an addition to the actual repo. If there's a better way to do this to make commenting easier, LMK!

**Comment by spmonahan** [comment]
I really appreciate how thorough the linked checklist is and that most of the questions I had while reading it were answered by continuing to read the doc :)

Having a11y testing be opened as a separate issue once a component is "done" is a good call as this checklist is not so much long but potentially time-consuming. In general that feels okay to me as one of our goals is to produce highly accessible components but I my initial feeling is that this checklist is a bit daunting. For example:

1. There is a large-ish testing matrix for screen readers that requires access to several different devices
2. Some of these tests require a fair bit of specialized knowledge of things like the ARIA spec and how different screen readers work (and how daily screen reader users actually use them vs. how a tester uses them)

Some questions I have:

1. Is this list prioritized? If not, can we prioritize it? I recognize that we want to test all of these things but I'm curious if it's valuable to test tricky/common issues or things that tend to have a higher value to users. Likewise, should we look at which screen reader/browser combos are most common and prioritize those? _OR_ is prioritization a bad thing here because we want to get to all of these things in all cases?
2. Do you envision the component author taking on this a11y testing or should it be assigned to someone else? Should all of the testing be handled by one person?
3. You mention that some of these tests may need to be done by someone with specific knowledge of ARIA/screenreaders/etc. It's fine that some people will be more knowledgeable of certain things than others and I realize this is a bit tangential to this RFC but what are some ways this knowledge can be diffused across the team?

**Comment by smhigley** [comment]
@spmonahan the process I had in mind would go something like:
1. The component author would do the tests they feel comfortable doing (most likely to be excluded are screen reader/aria tests)
2. Pair or hand off the remaining tests

There's no real reason the component author would need to be the one to test it, other than that they'd probably have the best instinct for catching permutations of props and children that could affect the accessibility of the control. There's no other reason one person couldn't go do high contrast tests for multiple components all in one go, even if they didn't author them.

I'm hoping pairing to do some of the more needs-specific-knowledge tests will spread that knowledge around more, but definitely up for other suggestions for how to approach those.

Is there anything you think would be good to add to make it less daunting and more of a "think of this in pieces, do what you can right now" type of checklist?

**Comment by spmonahan** [comment]
> Is there anything you think would be good to add to make it less daunting and more of a "think of this in pieces, do what you can right now" type of checklist?

I've been thinking about this some more and nothing is really coming to mind. This is a good checklist and if we take the mindset of "do what you can, get help were you can't" we should be able to work through it nicely.

I definitely want to become more knowledgeable of Windows screen readers so I love the idea of pairing on that!
